routing/src/adjacency.rs

```diff
@@ -0,0 +1,40 @@
+use crate::interface::IfIndex;
+use net::eth::mac::Mac;
+use std::collections::HashMap;
+use std::net::IpAddr;
+
+#[allow(dead_code)]
+pub struct Adjacency {
+    pub address: IpAddr,
+    pub mac: Mac,
+    pub ifindex: IfIndex,
```

Comment:
It is fine for the moment, but I doubt that we want all of the members to be public long term.